jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/AsyncContentProducer.java

```diff
@@ -292,11 +292,14 @@ private Content.Chunk produceChunk()
                         LOG.debug("channel has no new chunk {}", this);
                     return null;
                 }
+                else
+                {
+                    _servletChannel.getState().onContentAdded();
+                }
             }
 
-            assert _chunk != null;
             // Release the chunk immediately, if it is empty.
-            if (!_chunk.hasRemaining() && !_chunk.isTerminal())
+            if (_chunk != null && !_chunk.hasRemaining() && !_chunk.isTerminal())
```

Comment:
@sbordet can you review this change.
Looks like if we ever hit line 281 then `_chunk` will be null here.

Reply:
I think we need to avoid using isTerminal as a condition for releasing.  We should just release any chunk we are given.